src/source/RazorEngine.Core/Compilation/ReferenceResolver/UseCurrentAssembliesReferenceResolver.cs

```diff
@@ -22,10 +22,23 @@ public IEnumerable<CompilerReference> GetReferences(TypeContext context = null,
         {
             return CompilerServicesUtility
                    .GetLoadedAssemblies()
-                   .Where(a => !a.IsDynamic && File.Exists(a.Location) && !a.Location.Contains(CompilerServiceBase.DynamicTemplateNamespace))
+                   .Where(IsValidAssembly)
                    .GroupBy(a => a.GetName().Name).Select(grp => grp.First(y => y.GetName().Version == grp.Max(x => x.GetName().Version))) // only select distinct assemblies based on FullName to avoid loading duplicate assemblies
                    .Select(a => CompilerReference.From(a))
                    .Concat(includeAssemblies ?? Enumerable.Empty<CompilerReference>());
         }
+
+        private static bool IsValidAssembly(System.Reflection.Assembly a)
+        {
+            try
+            {
+                return !a.IsDynamic && File.Exists(a.Location) && !a.Location.Contains(CompilerServiceBase.DynamicTemplateNamespace);
+            }
+            catch
```

Comment:
Can you clarify which exception is thrown and only catch the specific one (and add a comment)?
Otherwise (if we want to catch all) we need some kind of logging here at the very least.

Reply:
In AF this part is throwing exception.